### ISSUE TITLE
Remove fastapi-users-db-sqlmodel dependency

### ISSUE THREAD

**Issue by tcompa** [fractal-analytics-platform/fractal-server#488]
Context: this question came up while working (with @mfranzon) on a [pydantic/sqlmodel/sqlalchemy issue](https://github.com/fractal-analytics-platform/fractal-server/issues/489) which is not yet clear. We don't know if the two issues are related, but in general this one here has to be considered at some point.

--

We depend on [fastapi-users-db-sqlmodel](https://github.com/fastapi-users/fastapi-users-db-sqlmodel), that currently (as of their 0.2.0 version) forces sqlalchemy version to be <=1.4.35. The reason is that 1.4.36 introduced a widespread issue, including in sqlmodel  - see [release notes](https://github.com/sqlalchemy/sqlalchemy/releases/tag/rel_1_4_36) and several issues:
* https://github.com/tiangolo/sqlmodel/issues/315
* https://github.com/sqlalchemy/sqlalchemy/discussions/7972#discussioncomment-2655517
* https://github.com/tiangolo/sqlmodel/pull/461

sqlmodel 0.0.7 [partly fixed the issue on their side](https://github.com/tiangolo/sqlmodel/pull/322) (but notice [another PR is still open](https://github.com/tiangolo/sqlmodel/pull/461)), and as of sqlmodel 0.0.8 (released in August 2022) their sqlalchemy constraint is `">=1.4.17,<=1.4.41"`.

In fastapi-users-db-sqlmodel there are several automatic dependabot PRs to relax the constraint, but [little activity](https://github.com/fastapi-users/fastapi-users-db-sqlmodel/pulls?q=is%3Apr+is%3Aclosed).

**We should verify if this package actively mantained, and otherwise we should find some way to replace it.**

Note that no other dependency of ours currently requires sqlalchemy<=1.4.35:
```
$ poetry show sqlalchemy

 name         : sqlalchemy                   
 version      : 1.4.35                       
 description  : Database Abstraction Library 

dependencies
 - greenlet !=0.4.17

required by
 - alembic >=1.3.0
 - fastapi-users-db-sqlmodel >=1.4,<1.4.36
 - sqlalchemy-utils >=1.3
 - sqlmodel >=1.4.17,<=1.4.41
```

**Comment by tcompa** [comment]
For the record, let's keep in mind that https://github.com/fastapi-users/fastapi-users-db-sqlalchemy appears as well maintained.

**Comment by tcompa** [comment]
The three classes that we are using from this package are
* [SQLModelUserDatabaseAsync](https://github.com/fastapi-users/fastapi-users-db-sqlmodel/blob/main/fastapi_users_db_sqlmodel/__init__.py#L146)
* [SQLModelBaseUserDB](https://github.com/fastapi-users/fastapi-users-db-sqlmodel/blob/main/fastapi_users_db_sqlmodel/__init__.py#L15)
* [SQLModelBaseOAuthAccount](https://github.com/fastapi-users/fastapi-users-db-sqlmodel/blob/main/fastapi_users_db_sqlmodel/__init__.py#L35)

If we proceed with a switch from UUID4 to autoincremental integer IDs (ref #560), then `SQLModelBaseUserDB` and `SQLModelBaseOAuthAccount` become irrelevant.

As per `SQLModelUserDatabaseAsync`, right now I find it is used [eight times across github](https://github.com/search?q=SQLModelUserDatabaseAsync&type=code), and also see [comments like these ones](https://github.com/fastapi-users/fastapi-users/discussions/861)
> Bear in mind though that we didn't officially release this DB adapter; as they are some problems with SQLModel regarding UUID (https://github.com/tiangolo/sqlmodel/issues/25). So you'll probably run into issues.
> ..
> There is one that should work, but don't plan to officially support it for now: https://github.com/fastapi-users/fastapi-users-db-sqlmodel

I think that we should quickly move on and explore the possibility of fully removing this dependency. If it impossible to replace `SQLModelUserDatabaseAsync`, then let's also consider that it is released under [MIT license](https://github.com/fastapi-users/fastapi-users-db-sqlmodel#license) and we could include that definition in our code base.

**Comment by caniko** [comment]
Hello, would you consider supporting `sqlmodel` for `fastapi-users`?

**Comment by jluethi** [comment]
We being the Fractal developers, not the `fastapi-users` developers. You may want to check with them directly for your question @caniko , because we just use that library :)

**Comment by caniko** [comment]
I considered the same @tcompa, thanks

**Comment by tcompa** [comment]
This is now ready as part of #660. The choices we made are:
1. We merged the relevant part of `SQLModelBaseUserDB` into our own `UserOAuth`
2. We kept `SQLModelBaseOAuthAccount` as is (bundled in fractal-server with attribution, under MIT license), since it is a very transparent model with a few simple (string) attributes.
3. We kept `SQLModelUserDatabaseAsync` as is (bundled in fractal-server with attribution, under MIT license), since we could only re-implement it in the exact same way. If (unlikely) fastapi-users is updated in a breaking way (e.g. requiring an additional `get_by_xxx` method), then the current adapter won't work and some users endpoints may be break. This is mitigated by the fact that we already test all relevant fastapi-users endpoints (see `test_security.py`), and notably:
    * whoami
    * register
    * userlist
    * get user
    * delete user
    * patch user
    * create (super)user

Side note: if we ever move away from SQLModel, towards sqlalchemy v2, then we could revert the current change and easily switch to the (better maintained) corresponding adapter: https://github.com/fastapi-users/fastapi-users-db-sqlalchemy

**Comment by tcompa** [comment]
Closed with #660
